sdks/java/io/hadoop/input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java

```diff
@@ -115,6 +113,14 @@
  * for both key and value. If you specify a translation, you will need to make sure the K or V of
  * the read transform match the output type of the translation.
  *
+ * <p>You will need to set appropriate InputFormat key and value class (i.e. "key.class" and
+ * "value.class") in Hadoop {@link Configuration}. Key or value object is cloned by encoding or
```

Comment:
For clarity, I would just remove the sentences starting "Key or value object..." and "Based on the the..."